pkg/webhook/genericsimpleevent.go

```diff
@@ -55,57 +50,46 @@ func (g *genericWebhook) Handle(c *gin.Context) {
 		return
 	}
 
-	gwData := &genericWebhookData{}
-
-	err = c.BindJSON(gwData)
+	payload, err := ioutil.ReadAll(c.Request.Body)
 	if err != nil {
-		log.Printf("Failed to convert POST data into JSON: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data"})
+		log.Printf("Failed to read body: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
 		return
 	}
+	defer c.Request.Body.Close()
 
-	body, err := ioutil.ReadAll(c.Request.Body)
-
+	// try to unmarshal Revision data, if they do exist
+	revision := &brigade.Revision{}
+	err = json.Unmarshal(payload, &revision)
 	if err != nil {
-		log.Printf("Failed to read body: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
+		log.Printf("Failed to convert POST data into JSON: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data - Invalid JSON"})
 		return
 	}
-	defer c.Request.Body.Close()
 
-	go g.notifyGenericWebhookEvent(proj, body, gwData)
-	c.JSON(200, gin.H{"status": "Success"})
+	go g.notifyGenericWebhookSimpleEvent(proj, payload, revision)
```

Comment:
Why is a separate Goroutine used here?

Reply:
it's based on [this](https://github.com/Azure/brigade/blob/master/pkg/webhook/dockerhub.go#L63). I suppose the original intention is that we should return a 200 immediately to the caller, since kube API might delay the Secret creation. However, I don't see any harm in making it a synchronous call. WDYT?

Reply:
Oh, yes, that makes sense - not entirely sure if we should directly return a 200, but it makes sense, and if we are to change the behaviour, we should do it at once, across gateways.

Marking this as resolved, thanks!